### PR TITLE
feat(select): add a disable modifier

### LINKED ISSUE #1985: PF4 Select: Add the ability to  disable the component
The consumer should be able to disable the Select component similar to the Dropdown.

### PR DESCRIPTION
**Pull request by boaz0**
closes #1985 

//cc @tlabaj

### CODE REVIEW

File: src/patternfly/components/Select/select.scss [hunk 1]
```diff
@@ -22,9 +22,11 @@
   --pf-c-select__toggle--active--BorderBottomColor: var(--pf-global--active-color--100);
   --pf-c-select__toggle--expanded--BorderBottomWidth: var(--pf-global--BorderWidth--md);
   --pf-c-select__toggle--expanded--BorderBottomColor: var(--pf-global--active-color--100);
+  --pf-c-select__toggle--disabled--BackgroundColor: var(--pf-global--disabled-color--300);
   --pf-c-select__toggle--m-plain--BorderColor: transparent;
   --pf-c-select__toggle--m-plain--Color: var(--pf-global--Color--200);
   --pf-c-select__toggle--m-plain--hover--Color: var(--pf-global--Color--100);
+  --pf-c-select__toggle--m-plain--disabled--Color: var(--pf-global--disabled-color--200);
```
Comment (mcoker) [inline src/patternfly/components/Select/select.scss]: This var is unused. I'll comment in the issue and see if we even want to support a plain variation of the select component.

**Comment by mcoker** [comment]
@mcarrano we have support for a "plain" variation of the select, which would be similar to a kebab dropdown, or a sort icon options menu. Looks like it was added in the [original custom select implementation](https://github.com/patternfly/patternfly-next/pull/1096/), but isn't documented or in use anywhere as far as I know. Do we want to support that as a variation in the select, or should we just remove that variation?

**Comment by mcarrano** [comment]
Not sure that I entirely understand the question @mcoker . Are you asking if the Select should support a variation? Not sure what this would be used for because by definition a select list to picking a value to show from a list of values.

**Comment by mcoker** [comment]
@mcarrano Yeah, in our "dropdown" menus (dropdown, options menu, etc), we support a "plain" variation that is like a kebab menu, and in the options menu we support a "plain text" variation where there is simply no border around the select toggle, like we use in pagination.

Screen Shot 2019-07-15 at 10 21 38 AM

Currently we have support for the plain variation of a select (in the code, but not in an example) but not the plain text. I was referring to the plain variation in this PR. I'm assuming we don't want to support a plain select component, but we do want to add support for the plain text variation?

**Comment by mcarrano** [comment]
I see. So this would be something that works like a dropdown but without the border. Do I understand that right @mcoker ? I'm struggling to think of a use case for that. If we want a text label, wouldn't we just use the standard dropdown/select? Can you think of a unique use case that requires this? I don't have a problem keeping this as an undocumented feature if it's there. If I'm still misunderstanding, let's discuss.

File: src/patternfly/components/Select/select.scss (hunk 1 above)
Comment (mcoker) [reply src/patternfly/components/Select/select.scss]: Hey @boaz0 we do want to support this variation, but you can remove this var. The disabled plain select should display the same as the regular disabled select.

File: src/patternfly/components/Select/select.scss [hunk 2]
```diff
@@ -144,6 +146,19 @@
   background-color: var(--pf-c-select__toggle--BackgroundColor);
   border: none;
 
+  &.pf-m-disabled,
+  &:disabled {
+    pointer-events: none;
+
+    &:not(.pf-m-plain) {
```
Comment (mcoker) [inline src/patternfly/components/Select/select.scss]: you can remove this selector, too. we want the plain variation to get the background color and `border: 0`

File: src/patternfly/components/Select/docs/select-single.md [hunk 3]
```diff
@@ -7,6 +7,7 @@ The single select should be used when the user is selecting an option from a lis
 | Attribute | Applied to | Outcome |
 | -- | -- | -- |
 | `role` or `aria` | `pf-c-select` |  accessibility notes. |
+| `disabled` | `.pf-c-select__toggle` | Disables the dropdown toggle and removes it from keyboard focus. |
```
Comment (christiemolloy) [inline src/patternfly/components/Select/docs/select-single.md]: I think you also want to add in the modifier class for disabled.

File: src/patternfly/components/Select/select.scss [hunk 4]
```diff
@@ -144,6 +145,17 @@
   background-color: var(--pf-c-select__toggle--BackgroundColor);
   border: none;
 
+  &.pf-m-disabled,
+  &:disabled {
+    pointer-events: none;
+
+    --pf-c-select__toggle--BackgroundColor: var(--pf-c-select__toggle--disabled--BackgroundColor);
```
Comment (mcoker) [inline src/patternfly/components/Select/select.scss]: Can you move the variable to the top? We typically list variables before declarations.

**Comment by mcoker** [comment]
@boaz0 can you make sure you have the latest from master? Still doesn't look like this pulled in the select plain example.

**Comment by boaz0** [comment]
@mcoker I fetched the latest updates. Can you make sure I didn't miss anything?
Thanks.

**Comment by boaz0** [comment]
@mcoker done!

**Comment by mcoker** [comment]
@boaz0 can you pull the latest from master and rebase? That should fix the failing build.

**Comment by boaz0** [comment]
Yes sure!

**Comment by boaz0** [comment]
Thanks a lot @mcoker for your help, thanks @christiemolloy for your feedback and thanks @redallen for merging :smile_cat:

**Pull request merged**